packages/wonder-blocks-dropdown/src/components/combobox.tsx

```diff
@@ -186,6 +186,7 @@ export default function Combobox({
         children: currentOptions,
         disabled,
         id: uniqueId,
+        onChange: (value) => handleChange(value),
```

Comment:
**note:** This is the main change in this PR. Instead of relying on `useEffect`, now we rely on the hook to call `onChange` so we can control when the consumer needs to act based on that change (instead of `useEffect` controlling that).